packages/compat/src/v1-addon.ts

```diff
@@ -429,6 +444,10 @@ export default class V1Addon {
     if (this.templateCompiler) {
       babelConfig.plugins.push(this.templateCompiler.inlineTransformsBabelPlugin());
     }
+
+    if (this.needsTypeScriptTransform) {
+      babelConfig.plugins.push(require.resolve('@babel/plugin-transform-typescript'));
```

Comment:
I didn't think this would be necessary, because ember-cli-babel adds the typescript plugin when it sees that ember-cli-typescript 4x is present?

Your fix above in `needsCustomBabel` should ensure that we allow ember-cli-babel to do its usual thing (other than the changes we make to its own settings like `compileModules: false`).

If this line is really needed to make it work, we should debug why because then I don't understand it.

Reply:
Also, if we do keep this line, we should probably guard against double insertion of the plugin. If anybody else has already inserted it, babel will throw when it sees the duplicate.

For an example of where we already do this, see https://github.com/embroider-build/embroider/blob/3d48872ece01cedb9dd81da754dc5d75a9052b21/packages/webpack/src/ember-webpack.ts#L500-L502

Reply:
Also, this is resolving the plugin relative to `@embroider/compat`, which doesn't have a dependency on `@babel/plugin-transform-typescript`. It needs to resolve relative to `ember-cli-babel`'s root, because AFAIK that is the package that depends on the plugin. But again, this is one more reason I think it's suspicious to add the plugin here -- I think ember-cli-babel should probably be doing it.

Reply:
You're totally right. All it needed was https://github.com/embroider-build/embroider/pull/529/commits/701009a54b39500fcffaac22bd142d21753218eb!

I'm not sure how I ended up down the rabbit hole where I thought I needed to have Embroider add the plugin. ¯\\\_(ツ)\_/¯